groups:
  LegFR:
    base_link: 
      - M
    end_link:
      - IMU2FR
    joints: 
      - SeFR_M
      - ThFR_SeFR
      - ShFR_ThFR
    limits:
      SeFR_M:
        min: -0.7854
        max: 0.7854
      ThFR_SeFR:
        min: -0.7854
        max: 0.7854
      ShFR_ThFR:
        min: -0.7854
        max: 0.7854
    lengths: 
      - 0.1762
      - 0.181
      - 0.170

  LegFL:
    base_link: 
      - M
    end_link:
      - IMU2FL
    joints:
      - SeFL_M
      - ThFL_SeFL
      - ShFL_ThFL
    limits:
      SeFL_M:
        min: -0.7854
        max: 0.7854
      ThFL_SeFL:
        min: -0.7854
        max: 0.7854
      ShFL_ThFL:
        min: -0.7854
        max: 0.7854
    lengths: 
      - 0.1762
      - 0.181
      - 0.170

  LegBR:
    base_link: 
      - M
    end_link:
      - IMU2BR
    joints:
      - SeBR_M
      - ThBR_SeBR
      - ShBR_ThBR
    limits:
      SeBR_M:
        min: -0.7854
        max: 0.7854
      ThBR_SeBR:
        min: -0.7854
        max: 0.7854
      ShBR_ThBR:
        min: -0.7854
        max: 0.7854
    lengths: 
      - 0.1563
      - 0.181
      - 0.170

  LegBL:
    base_link: 
      - M
    end_link:
      - IMU2BL
    joints:
      - SeBL_M
      - ThBL_SeBL
      - ShBL_ThBL
    limits:
      SeBL_M:
        min: -0.7854
        max: 0.7854
      ThBL_SeBL:
        min: -0.7854
        max: 0.7854
      ShBL_ThBL:
        min: -0.7854
        max: 0.7854
    lengths:
      - 0.1563
      - 0.181
      - 0.170
      
solvers:
  grid:
    max_iter: 5000
    delta: 1e-3
    radius: 5e-1
    enable_refinement: true
    max_refinement_iter: 10
    random_seed: 1 # (-1: time) | (n: predictable seed)  
